### ISSUE TITLE
Nested to_hash in Tire::Results::Item

### ISSUE THREAD

**Issue by thedaniel** [karmi/retire#339]
When Tire::Results::Item wraps a nested hash like

```
{
 username: bob,
 avatars: {
   big: 'http://foo',
   small: 'http://bar
  }
}
```

both the outer hash and inner hash (`avatars` in this example) are cast to Tire::Results::Item. This is awesome for dot notation access to nested hashes.

Unfortunatey, if you want to manually serialize this with e.g. `to_json`, only the outer hash is recast to Hash, leaving the inner has still as an Item. This causes to_json to crash like:

```
ruby-1.9.2-p180 :021 > h.to_json
ArgumentError: wrong number of arguments (2 for 1)
    from /Users/daniel/.rvm/gems/ruby-1.9.2-p180@ng2/gems/tire-0.4.2/lib/tire/results/item.rb:75:in `to_json'
```

Basically, i think that to_hash and to_json should recursively un-cast from Item back to Hash, not just do the outermost layer.

**Comment by thedaniel** [comment]
After looking at my data that caused the problem, i think it might not be nested hashes, but Items inside arrays, like

```
{
  foo: 'bar',
  baz: [Item(), Item()],
}
```

**Comment by karmi** [comment]
Hmm, this sounds like a nasty bug. Could you have a look into unit tests for `Tire::Results::Item` if you'd be able to write a failing test?

**Comment by thedaniel** [comment]
whoops, just saw this - i had notifications turned off for some reason. i will make some time to try to write a failing test this week sometime

**Comment by azitabh** [comment]
Any update on this?

**Comment by karmi** [comment]
Should be fixed in karmi/tire@4f2cbd4

**Comment by karmi** [comment]
@azitabh @thedaniel Thanks for your feedback, hope your problem is solved. Closing for now.
